skipper-py/src/bot.py

```diff
@@ -177,6 +168,18 @@ async def run(self):
                 # If there are no routes, continue
                 if not transaction.routes:
                     continue
+
+                # Update reserves once per potential backrun list
```

Comment:
One trade-off with this method is that given `backrun_list`, previously we would only query the node once (with many many requests in parallel) whereas this new logic does this iteratively for each tx. I wonder if a more optimal design would be to get all the unique pools from routes across the entire backrun list such that the update only happens once.

This may be a minor optimization in the end though that I'm down to just note here but not block on, and we can always optimize more if we need to.